bumble/gatt_client.py

```diff
@@ -332,8 +344,9 @@ async def discover_services(self, uuids=None):
                         '!!! unexpected error while discovering services: '
                         f'{HCI_Constant.error_name(response.error_code)}'
                     )
-                    # TODO raise appropriate exception
-                    return
+                    raise Exception(
```

Comment:
suggest: create a more specific exception class. Maybe something like `ATT_Error(response.error_code)`, like we do for `HCI_Error` (i.e with an error code, error namespace and error name).